src/clearskies/columns/has_many.py

```diff
@@ -463,12 +463,32 @@ def __get__(self, instance: Model, cls: type[Model]) -> Model:
     def __get__(self, model, cls):
         if model is None:
             self.model_class = cls
-            return self  # type:  ignore
+            return self  # type: ignore
 
         # this makes sure we're initialized
         if "name" not in self._config:  # type: ignore
             model.get_columns()
 
+        # The data will be in model._data[column_name] as a list of child models
+        raw_data = model.get_raw_data()
+
+        if self.name in raw_data and isinstance(raw_data[self.name], list):
+            preloaded_models = raw_data[self.name]
+
+            # Check if this is pre-loaded relationship data
+            # Empty lists are valid (no related records)
+            # Non-empty lists should contain model instances
+            if not preloaded_models or all(hasattr(m, "__class__") and hasattr(m, "_data") for m in preloaded_models):
+                # Create a Models instance with pre-loaded data
+                # Use the parent class to get the base query Model instance
+                children = super().__get__(model, cls)
+                # Set the pre-loaded models on the QUERY object (not the Model)
+                # The backend checks query._models in its records() method
+                if children._query:
```

Comment:
Hmm, but now that I'm looking at this I'm not 100% sure what you have going on.  It generally should be as simple as:

```
if self.name in model._transformed_data:
    return model._transformed_data[self.name]
```

And then below, anywhere you have a `return children`  you would replace it with:

```
model._transformed_data[self.name] = children
return children
```

But, I wonder if I'm missing something you were going for here?